File: Data Structure II/python/Day I/majority_element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        map={};
        for i in range(len(nums)):
          if nums[i] in map:
            map[nums[i]]+=1
          else:
            map[nums[i]]=1
        for i in map:
            if map[i]>len(nums)//2:
                return i;
        return -1; 

File: Data Structure II/cpp/Day I/MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> map;
        for(int i=0;i<nums.size();i++){
            map[nums[i]]++;
        }
        for(auto i : map){
            if(i.second>nums.size()/2){
                return i.first;
            }
        }
        return -1; 
    }
};